### DISCUSSION TITLE
Page names and component names interfere when using auto import

### DISCUSSION THREAD

**Discussion by jerryjappinen** [nuxt/nuxt#17218] [Questions]
### Example use case

I'm creating documentation web site for a component library. I am auto importing this component library as a Nuxt module, and for each component I want to create a page that documents its behavior. I don't want to use Storybook or other similar solutions this time.

So, this will **not** work:

```
components/
  MyComponent.vue
pages/
  components/
    MyComponent.vue
  index.vue
```

If you are using auto imports, have a component named `MyComponent.vue` and also have a page under `pages/` with the route `MyComponent`, Nuxt will get clogged up and throw a maximum call stack error.

---

### Mimimal repro

See [github.com/jerryjappinen/nuxt-page-name-conflict](https://github.com/jerryjappinen/nuxt-page-name-conflict)

---

### Attempted workarounds

You could say I should prefix the component names somehow, but that doesn't help. I want the route name to exactly match the component name, so that for the end user last part of the route matches exactly what they need when they `import` that component.

I also tried creating a dynamic route so that I could keep the documentation components with custom names elsewhere. This also doesn't seem to work though, since Nuxt 3 requires the use of `resolveComponent()` that does not support dynamic component names.

Burying the page components under subdirectories does not help. Auto importer still goes haywire - I think it just looks at the filename.

---

### Thoughts and questions

Since the page component names are determined by desired route names, and thus have a direct impact on end-user experience, I feel like Nuxt has an awkward limitation here. This specific use case is somewhat specific, but I can't find a good workaround either and I can see that this will be an issue every now and then.

Questions:

1. What would be the appropriate, most sensible workaround here? All I can think of is ditching the auto imports and sacrificing my own DX here. I feel like this isn't really the intended _Nuxt_ way of doing things though.
2. Wouldn't the auto importer be much more robust, if it could prefix or in some other way treat the page components as separate from other components? I can't think of a scenario why I'd ever use a page component in another component, and it's probably not intended either. It's hard for me as a developer to understand, why this is an issue.

I feel like this is a bug, but I wonder if there's something I'm missing or if it sparks any discussion.

**Comment by nathanchase** [comment]
You might try naming your pages '____Page' (aka "UserPage", "LoginPage", etc.) using [the second script tag method](https://github.com/nuxt/framework/discussions/5985) to avoid name collisions.
```js
// /pages/user/index.vue
<template>
   <User />
</template>

<script>
export default {
  name: 'UserPage',
};
</script>

<script scoped lang="ts">
...
</script>
```

**Reply by jerryjappinen** [reply]
This did occur to me as well and I tried it, but unfortunately it doesn't work. You can try it in the repo I linked to above.

Why it doesn't work - another good question. You would expect it to I guess.

**Comment by Thomaash** [comment]
Just in case anyone ends up here from search (like I did today), this is a Vue feature called recursive components (https://vuejs.org/api/sfc-script-setup#recursive-components). It has nothing to do with Nuxt nor auto imports. Adding `import { MyComponent } from "#components";` should work as a workaround.
